java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -252,31 +283,82 @@ public abstract class BaseClient
      * @param <T> Client type.
      * @return a Future to connect and return a RedisClient.
      */
-    protected static <T> CompletableFuture<T> CreateClient(
-            BaseClientConfiguration config,
-            BiFunction<ConnectionManager, CommandManager, T> constructor) {
+    protected static <T extends BaseClient> CompletableFuture<T> CreateClient(
+            @NonNull BaseClientConfiguration config, Supplier<T> constructor) {
         try {
             ThreadPoolResource threadPoolResource = config.getThreadPoolResource();
             if (threadPoolResource == null) {
                 threadPoolResource =
                         ThreadPoolResourceAllocator.getOrCreate(Platform.getThreadPoolResourceSupplier());
             }
-            ChannelHandler channelHandler = buildChannelHandler(threadPoolResource);
+            MessageHandler messageHandler = buildMessageHandler(config);
+            ChannelHandler channelHandler = buildChannelHandler(threadPoolResource, messageHandler);
             ConnectionManager connectionManager = buildConnectionManager(channelHandler);
             CommandManager commandManager = buildCommandManager(channelHandler);
             // TODO: Support exception throwing, including interrupted exceptions
             return connectionManager
                     .connectToRedis(config)
-                    .thenApply(ignore -> constructor.apply(connectionManager, commandManager));
+                    .thenApply(
+                            ignored ->
+                                    buildClient(
+                                            constructor,
+                                            connectionManager,
+                                            commandManager,
+                                            messageHandler,
+                                            config.getSubscriptionConfiguration()));
         } catch (InterruptedException e) {
-            // Something bad happened while we were establishing netty connection to
-            // UDS
+            // Something bad happened while we were establishing netty connection to UDS
             var future = new CompletableFuture<T>();
             future.completeExceptionally(e);
             return future;
         }
     }
 
+    /**
+     * Tries to return a next pubsub message.
+     *
+     * @throws WrongConfiguration If client is not subscribed to any channel or if client configured
+     *     with a callback.
+     * @return A message if any or <code>null</code> if there are no unread messages.
+     */
+    public Message tryGetPubSubMessage() {
```

Comment:
why "try"?  what is the purpose of this method vs `getPubSubMessage`?

Reply:
I followed python client namings: https://github.com/aws/glide-for-redis/pull/1616/files#diff-f0f69819afe466c0f3e8afd4ea2fd31ca4f451373fa8ed1351f48473a1c39633R5042
```py
async def get_pubsub_message(self) -> PubSubMsg:
def try_get_pubsub_message(self) -> Optional[PubSubMsg]:
```